src/main/java/com/frameworkium/core/ui/ExtraExpectedConditions.java

```diff
@@ -21,6 +21,39 @@
 public class ExtraExpectedConditions {
 
     /**
+     * An expectation for checking that N elements (from a list) are present on the web page that match the locator
+     * are visible. Visibility means that the elements are not only displayed but also have a height
+     * and width that is greater than 0.
+     *
+     * @param elementsToCheck how many elements in a <code>List&lt;WebElement&gt;</code> to check
+     * @param elements list of WebElements
+     * @return the list of WebElements once they are located
+     */
+    public static ExpectedCondition<List<WebElement>> visibilityOfNElements(
+            final List<WebElement> elements,int elementsToCheck) {
+
+        int finalElementsToCheck = elementsToCheck == 0 ? elements.size() : elementsToCheck;
+
+        return new ExpectedCondition<List<WebElement>>() {
+            @Override
+            public List<WebElement> apply(WebDriver driver) {
+                for (int i = 0; i < finalElementsToCheck; i++) {
+                    WebElement element = elements.get(i);
+                    if (!element.isDisplayed()) {
+                        return null;
```

Comment:
``
elements.stream()
 .filter(WebElement::isDisplayed)
              .limit(finalElementsToCheck)
             .collect(Collectors.toList());

Reply:
only problem, iirc, with this is that it needs to return null to indicate "false"

I was thining:

```java
boolean atLeastNVisible = CollectionUtils.isNotEmpty(elements)
                            && elements.stream()
                            .limit(atLeast == 0 ? elements.size() : atLeast)
                            .allMatch(WebElement::isDisplayed);

                    return atLeastNVisible ? elements : null;
```

Reply:
Perhaps?
```
              return  elements.stream()
                        .filter(WebElement::isDisplayed)
                        .limit(finalElementsToCheck)
                        .collect(Collectors.toList())
                        .size() == finalElementsToCheck ? elements : null;

Reply:
You would need the limit before the filter, but that could also work.

Reply:
or, maybe not.. that's an interesting ambiguity of `atLeast` I guess.. I assumed it was the first `n`, but actually it could be any `n`

Reply:
looks like we are calling isDisplayed on all elements here

Reply:
That would warrant for another annotation value?
```atLeast``` vs ```exactly```

Reply:
or even `atLeast`, `exactly`, and `firstN`... but actually `firstN` seems too niche. Ok, so, new feature request, `exactly` :D

Reply:
@noronhaa it will only call `isDisplayed` until it reached the limit, good ol' lazy streams :)

Reply:
so it won't call isDisplayed on all elements and then after that limit n of those elements to then collect etc?

Reply:
nope, it will only call as many times as it needs until it reached the limit after the filter, i.e. until the number of displayed items reaches the limit (or it runs out of elements).

Also, slight improvement
```java
return elements.stream()
        .filter(WebElement::isDisplayed)
        .limit(finalElementsToCheck)
        .count() == finalElementsToCheck ? elements : null;
```